lib/wickedbase.pm

```diff
@@ -457,6 +457,52 @@ sub validate_macvtap {
     validate_script_output("cat $macvtap_log", sub { m/Success listening to tap device/ });
 }
 
+sub setup_bond() {
+    my ($self, $mode, $iface0, $iface1) = @_;
+    my $ifsut = $mode == "balance-rr" ? "bond0-rr"
+      : $mode == "active-backup" ? "bond0-ab"
+      : $mode == "balance-xor"   ? "bond0-xor"
+      : $mode == "broadcast"     ? "bond0-bc"
+      : $mode == "802.3ad"       ? "bond0-ieee"
+      : $mode == "balance-tlb"   ? "bond0-tlb"
+      : $mode == "balance-alb"   ? "bond0-alb"
+      :                            die("Unsupported bond mode $mode");
+
+    my $cfg_bond0 = '/etc/sysconfig/network/ifcfg-bond0';
+    my $cfg_ifc0  = '/etc/sysconfig/network/ifcfg-' . $iface0;
+    my $cfg_ifc1  = '/etc/sysconfig/network/ifcfg-' . $iface1;
+    $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug', $cfg_ifc0);
+    $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug', $cfg_ifc1);
+    $self->get_from_data('wicked/bonding/ifcfg-' . $ifsut,  $cfg_bond0);
+    file_content_replace($cfg_bond0, ipaddr4 => $self->get_ip(type => 'host', netmask => 1), ipaddr6 => $self->get_ip(type => 'host6', netmask => 1), iface0 => $iface0, iface1 => $iface1);
+
+    $self->wicked_command('ifup', 'all');
+}
+
+
+sub setup_team() {
+    my ($self, $mode, $iface0, $iface1) = @_;
+
+    my $cfg_team0 = '/etc/sysconfig/network/ifcfg-team0';
+    my $cfg_ifc0  = '/etc/sysconfig/network/ifcfg-' . $iface0;
+    my $cfg_ifc1  = '/etc/sysconfig/network/ifcfg-' . $iface1;
+    $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc0);
+    $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc1);
+    $self->get_from_data('wicked/teaming/ifcfg-team0-' . $mode, $cfg_team0);
+    file_content_replace($cfg_team0, ipaddr4 => $self->get_ip(type => 'host', netmask => 1), ipaddr6 => $self->get_ip(type => 'host6', netmask => 1), iface0 => $iface0, iface1 => $iface1);
+
+    $self->wicked_command('ifup', 'all');
+}
+
+
+sub test_aggregate() {
```

Comment:
```suggestion
sub validate_interfaces() {
```

WDYT ?

Reply:
my problem with `test_aggregate` that first time I have read it like "aggregate test cases" :)

Reply:
ok, makes more sense as we use "validate" in other places too